restapi/src/main/java/io/stargate/web/docsapi/service/QueryExecutor.java

```diff
@@ -57,30 +76,120 @@ public QueryExecutor(DataStore dataStore) {
       int pageSize,
       ByteBuffer pagingState,
       ExecutionContext context) {
+    return queryDocs(keyDepth, ImmutableList.of(query), pageSize, pagingState, context);
+  }
+
+  /**
+   * Runs the provided queries in parallel and merges their result sets according to the selected
```

Comment:
ohh this is so nice, we needed this kind of detailed explanation what s possible what not